version: '3.8'

services:
  oak-d:
    build:
      context: .
      dockerfile: oak-d/Dockerfile
    container_name: oak-d
    network_mode: host
    privileged: true
    environment:
      - ROS_DOMAIN_ID=0
      - DISPLAY=:0
      - RMW_IMPLEMENTATION=rmw_zenoh_cpp
      - ZENOH_ROUTER_CHECK_ATTEMPTS=10  # Пытаемся подключиться к router 10 раз
      - RUST_LOG=zenoh=info  # Логирование Zenoh для отладки
      # Zenoh endpoints - подключение к роутеру Main Pi по Gigabit Ethernet
      - ROS_AUTOMATIC_DISCOVERY_RANGE=SUBNET
      - ZENOH_CONNECT=tcp/10.1.1.10:7447
    volumes:
      - ./config:/config
      - ./config/zenoh_session_config.json5:/config/zenoh_session_config.json5
      - /dev/bus/usb:/dev/bus/usb
      - /dev/shm:/dev/shm
    restart: unless-stopped
    
  apriltag:
    build:
      context: .
      dockerfile: apriltag/Dockerfile
    container_name: apriltag
    network_mode: host
    environment:
      - ROS_DOMAIN_ID=0
      - RMW_IMPLEMENTATION=rmw_zenoh_cpp
      - ZENOH_ROUTER_CHECK_ATTEMPTS=10
      - RUST_LOG=zenoh=info
      # Zenoh endpoints - подключение к роутеру Main Pi по Gigabit Ethernet
      - ROS_AUTOMATIC_DISCOVERY_RANGE=SUBNET
      - ZENOH_CONNECT=tcp/10.1.1.10:7447
    volumes:
      - ./config:/config
      - ./config/zenoh_session_config.json5:/config/zenoh_session_config.json5
      - /dev/shm:/dev/shm
    depends_on:
      oak-d:
        condition: service_started
    restart: unless-stopped